### ISSUE TITLE
iallocator fails for diskless instances

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#463]
Originally reported of Google Code with ID 407.

```
# gnt-instance add -t diskless -o busybox ultrt-gnt5.hot.corp.google.com
Failure: command execution error:
Failure while running the iallocator script: iallocator module 'hail' failed: exited with exit code 1, output 'Error: invalid data for instance 'ultrt-gnt5.hot.corp.google.com': key 'disk_space_total': Unable to read Int
'

Moreover, and even worse, if a diskless instance exists (created manually, without iallocator), iallocator fails forever on:

# gnt-instance add -t drbd -s 1G -o busybox ultrt-gnt3.hot.corp.google.com
Failure: command execution error:
Failure while running the iallocator script: iallocator module 'hail' failed: exited with exit code 1, output 'Error: invalid data for instance 'ultrt-gnt4.hot.corp.google.com': key 'disk_space_total': Unable to read Int
'

```
Originally added on 2013-04-04 09:25:38 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-05 08:20:36 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-12 08:56:25 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
fixed by 9e946416b10e4a070fbd07313df57
```
Originally added on 2013-04-12 09:00:52 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:49 +0000 UTC.

Changed State: Released
